docs/blog/09-25-2024-introducing-the-federation-api.md

```diff
@@ -0,0 +1,404 @@
+import * as d3 from "d3";
+import {onMounted, computed, defineAsyncComponent, Ref, ref, watch} from "vue"
+import { inBrowser } from 'vitepress';
+
+import Api from "@/src/api";
+import { ModelsSample as Sample } from "@/src/api/core";
+
+const ApiCallBlock = inBrowser
+  ? defineAsyncComponent(() => import('@/theme/components/api-call/Block.vue'))
+  : () => null;
+
+import * as Graph from "./09-25-2024-introducing-the-federation-api/graph";
+import GraphPlaceholder from "./09-25-2024-introducing-the-federation-api/GraphPlaceholder.vue";
+import { useDataStore } from "./09-25-2024-introducing-the-federation-api/store";
+
+let api = new Api();
+const data = useDataStore();
+const samples: Ref<Sample[]> = ref([]);
+
+const prepareByCount = (data: any): any => {
+  let result = [];
+
+  for (const [server, files] of Object.entries(data)) {
+    if (files === null) {
+      continue;
+    }
+
+    for (let entry of files.values) {
+      entry.label = server;
+      result.push({
+        group: server,
+        value: entry.value,
+        count: entry.count
+      });
+    }
+  }
+
+  return result.filter((d) => d.count > 0);
+}
+
+const plotSubjectByCount = async (id: string, title: string, field: string, options: any) => {
+  let data = prepareByCount(await api.countSubjectsBy(field));
+  Graph.bar(id, title, data, options);
+}
+
+const plotSampleByCount = async (id: string, title: string, field: string, options: any) => {
+  let data = prepareByCount(await api.countSamplesBy(field));
+  Graph.bar(id, title, data, options);
+}
+
+const plotFileByCount = async (id: string, title: string, field: string, options: any) => {
+  let data = prepareByCount(await api.countFilesBy(field));
+  Graph.bar(id, title, data, options);
+}
+
+onMounted(async() => {
+  // api.samples((name, kind, values, complete) => {
+  //   if (values !== null) {
+  //     samples.value.push(...values);
+  //   }
+  // });
+
+  plotFileByCount(
+    "#fileByType",
+    "Files by Type",
+    "type",
+    {
+      scale: 1.05
+    }
+  );
+
+  plotSampleByCount(
+    "#sampleByStrategy",
+    "Samples by Library Strategy",
+    "library_strategy"
+  );
+
+  plotSubjectByCount(
+    "#subjectByRace",
+    "Subjects by Race",
+    "race",
+    {
+      margin: {
+        bottom: 150,
+      },
+    }
+  );
+});
+
+const osteosarcomas = computed(() => {
+  return samples.value.filter((s) => {
+    let diagnosis = s.metadata?.diagnosis?.value || null;
+
+    return diagnosis !== null &&
+           diagnosis !== undefined &&
+           diagnosis.toLowerCase().includes("osteosarcoma");
+  });
+});
+
+const neuroblastomas = computed(() => {
+  return samples.value.filter((s) => {
+    let diagnosis = s.metadata?.diagnosis?.value || null;
+
+    return diagnosis !== null &&
+           diagnosis !== undefined &&
+           diagnosis.toLowerCase().includes("neuroblastoma");
+  });
+})
+
+watch([osteosarcomas, neuroblastomas], ([osteosarcomas, neuroblastomas], _) => {
```

Comment:
```suggestion
watch([osteosarcomas, neuroblastomas], ([osteosarcomas, neuroblastomas]) => {
```
I think if you don't need the old watcher values, you can omit them from the function signature.